packages/nx/src/tasks-runner/utils.ts

```diff
@@ -69,6 +69,13 @@ export function expandDependencyConfigSyntaxSugar(
 
   // Support for both `project:target` and `target:with:colons` syntax
   const [maybeProject, ...segments] = splitByColons(targetString);
+
+  // if no additional segments are provided, then the string references
+  // a target of the same project
+  if (!segments.length) {
+    return { target: maybeProject };
+  }
```

Comment:
```suggestion
  if (segments.length == 1) {
    return { target: maybeProject };
  }
```

This condition as written would never be hit, since there is always at least 1 segment (that just happens to be the target name most of the time). 

It's OK IMHO to assume it's a target name when only one segment is present, so updating the condition like above should suffice.

Reply:
Also for what it's worth - I tested the change as I originally wrote it in my monorepo (via `patch-package`) and it did fix the issue I was experiencing.